### PR TITLE
Avoid invoking remote EnC service for documents that do not support remote workspace

### PR DESCRIPTION
**Pull request by tmat**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1438496/

### CODE REVIEW

**Comment by tmat** [comment]
@davidwengier PTAL

File: src/EditorFeatures/Test/EditAndContinue/EditAndContinueWorkspaceServiceTests.cs [hunk 1]
```diff
@@ -746,55 +746,73 @@ public async Task DesignTimeOnlyDocument_Dynamic()
         }
 
         [Theory]
-        [InlineData(true)]
-        [InlineData(false)]
-        public async Task DesignTimeOnlyDocument_Wpf(bool delayLoad)
+        [CombinatorialData]
+        public async Task DesignTimeOnlyDocument_Wpf([CombinatorialValues(LanguageNames.CSharp, LanguageNames.VisualBasic)] string language, bool delayLoad, bool open)
```
Comment (tmat) [inline src/EditorFeatures/Test/EditAndContinue/EditAndContinueWorkspaceServiceTests.cs:750]: These test changes are unrelated to the remote issue, just increasing coverage.

**Review by davidwengier — APPROVED** [review]

**Comment by tmat** [comment]
@ManishJayaswal for approval
